src/SDKs/Storage/Storage.Tests/Tests/StorageAccountTests.cs

```diff
@@ -1288,5 +1288,121 @@ public void StorageAccountCreateWithHttpsOnlyTest()
             }
         }
+
+        [Fact]
+        public void StorageAccountCMKTest()
+        {
+            var handler = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+
+            using (MockContext context = MockContext.Start(this.GetType().FullName))
+            {
+                var resourcesClient = StorageManagementTestUtilities.GetResourceManagementClient(context, handler);
+                var storageMgmtClient = StorageManagementTestUtilities.GetStorageManagementClient(context, handler);
+                var keyVaultMgmtClient = StorageManagementTestUtilities.GetKeyVaultManagementClient(context, handler);
+
+                // Create storage account with hot
+                string accountName = TestUtilities.GenerateName("sto");
+                var rgname = "testcmk3";
+                string keyName = "testkey3";
+                string vaultName = "testkeyvaultcmk3";
+
+                var parameters = StorageManagementTestUtilities.GetDefaultStorageAccountParameters();
+                parameters.Identity = new Identity {};
+                var account = storageMgmtClient.StorageAccounts.Create(rgname, accountName, parameters);
+
+                StorageManagementTestUtilities.VerifyAccountProperties(account, false);
+                Assert.NotNull(account.Identity);
+
+                var keyVault = keyVaultMgmtClient.Vaults.Get(rgname, vaultName);
+
+                keyVault.Properties.AccessPolicies.Add(new Microsoft.Azure.Management.KeyVault.Models.AccessPolicyEntry
+                {
+                    TenantId = System.Guid.Parse(account.Identity.TenantId),
+                    ObjectId = account.Identity.PrincipalId,
+                    Permissions = new Microsoft.Azure.Management.KeyVault.Models.Permissions(new List<string> { "wrapkey", "unwrapkey" })
+                });
+
+                keyVault = keyVaultMgmtClient.Vaults.CreateOrUpdate(rgname, vaultName, parameters: new Microsoft.Azure.Management.KeyVault.Models.VaultCreateOrUpdateParameters
+                {
+                    Location = keyVault.Location,
+                    Properties = keyVault.Properties
+                });
+
+                // Enable encryption.
+                var updateParameters = new StorageAccountUpdateParameters
+                {
+                    Encryption = new Encryption
+                    {
+                        Services = new EncryptionServices { Blob = new EncryptionService { Enabled = true }, File = new EncryptionService { Enabled = true } },
+                        KeySource = "Microsoft.Keyvault",
+                        KeyVaultProperties =
+                            new KeyVaultProperties
+                            {
+                                KeyName = keyName,
+                                KeyVaultUri = keyVault.Properties.VaultUri,
+                                KeyVersion = "dccda185f55f4c34bcd2b86cc1bfff78"
```

Comment:
This is related to that pre-created KeyVault key. This is valid with that key vault key.